src/libvfcinstrument/libVFCInstrument.cpp

```diff
@@ -248,15 +275,14 @@ struct VfclibInst : public ModulePass {
       if (size > 1) {
         res = VectorType::get(res, size);
       }
-      Constant *hookFunc =
-          M.getOrInsertFunction(mcaFunctionName, res, Builder.getInt32Ty(),
-                                opType, opType, (Type *)0);
+      _LLVMFunctionType hookFunc = GET_OR_INSERT_FUNCTION(
+          M, mcaFunctionName, res, Builder.getInt32Ty(), opType, opType);
       newInst = CREATE_CALL3(hookFunc, Builder.getInt32(FCI->getPredicate()),
                              FCI->getOperand(0), FCI->getOperand(1));
       newInst = Builder.CreateIntCast(newInst, retType, true);
     } else {
-      Constant *hookFunc = M.getOrInsertFunction(mcaFunctionName, retType,
-                                                 opType, opType, (Type *)0);
+      _LLVMFunctionType hookFunc =
+          GET_OR_INSERT_FUNCTION(M, mcaFunctionName, retType, opType, opType);
```

Comment:
Why are we passing opType twice to this macro ?

Reply:
Because it is the type of the arguments. Since we have two arguments, we have to pass two `opType`.

Reply:
Ok sorry, I misread.